### DISCUSSION TITLE
Creating multiple db files with version

### DISCUSSION THREAD

**Discussion by osecen** [xerial/sqlite-jdbc#969] [Q&A]
We are using this library in one of our services and with the code change to add uuid to tmp files, our service started creating db of the form `sqlite-jdbc-tmp-3ca14337-a83f-4281-b8fb-a79f32aa69d0.db` continuously and the service runs out of disk space. With [this change](https://github.com/xerial/sqlite-jdbc/commit/edb4b8adc2447bc04e05b9b908195a4bc7926242), the old db is never deleted.  
I would like to confirm my thought process and explore ways to remediate this issue.

Thanks,
Onder

**Comment by gotson** [comment]
The change you refer to does not create the db but copy the dll to a temp dir.

**Reply by osecen** [reply]
Yes, our tmp directory gets flooded with those files and the service runs out of disk space.

**Reply by gotson** [reply]
my bad, this is not about temporary dll, but when you read a database from a `:resource:`.

can you clarify your use case maybe?

**Reply by osecen** [reply]
We have a CSV file and a sqlite database of that CSV file in the codebase. During runtime, server uses DriverManager from java.sql to get a Connection to the sqlite db on the server. During that call, since we are using a new filename with uuid, old files are ignored and the sqlite db is copied to a new db in tmp folder. In the old code, since the hashCode was the same, we didn't have this issue.

**Reply by gotson** [reply]
Can you explain how you load the db? Why do you use `:resource:` if you have the file locally?

**Reply by osecen** [reply]
The db file is located in the same jar with the code that is using this driver so it is not in the local file system. What is the proper way of using the db from the jar?
The current call has this format for db   `jdbc:sqlite::resource:com/myorg/data/mydatabase.db`

**Reply by gotson** [reply]
You should consider extracting yourself to a location you know has enough space.
